modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -4629,3 +4636,179 @@ def from_dataframe(cls, df: "ProtocolDataframe") -> "PandasDataframe":
+                            data = pandas.concat(
+                                [data, pandas.Series([fill_value] * diff)]
+                            )
+                    else:
+                        data = data.reindex(self_idx, fill_value=fill_value)
+                elif use_map and is_list_like(data):
+                    use_map = False
+                new_case.append(data)
+            new_caselist.append(tuple(new_case))
+
+        if modin_lists[0] or modin_lists[1]:
+            # Copartition modin frames
+            use_map = False
+            columns = self.columns
+            column_widths = [1]
+            for modin_list, fill_value in zip(modin_lists, fill_values):
+                _, list_of_right_parts, joined_index, row_lengths = self._copartition(
+                    Axis.ROW_WISE.value,
+                    modin_list,
+                    how="left",
+                    sort=False,
+                    fill_value=fill_value,
+                )
+                modin_list.clear()
+                modin_list.extend(
+                    self.__constructor__(
+                        part,
+                        joined_index,
+                        columns,
+                        row_lengths,
+                        column_widths,
+                    )
+                    for part in list_of_right_parts
+                )
+
+            # Replace modin frames with copartitioned
+            caselist = new_caselist
+            new_caselist = []
+            for i in range(2):
+                modin_lists[i] = iter(modin_lists[i])
+            for case_tuple in caselist:
+                new_case = tuple(
+                    next(modin_list) if isinstance(data, cls) else data
+                    for data, modin_list in zip(case_tuple, modin_lists)
+                )
+                new_caselist.append(new_case)
+
+        # If all the conditions are callable and the replacements are either
+        # callable or scalar, use map().
+        if use_map:
+            return self.map(func=remote_fn, func_args=[name, new_caselist], lazy=True)
+
+        # Get the chunk of data corresponding the the specified partition
+        def map_data(
+            part_idx,
+            part_len,
+            data,
+            data_offset,
+            fill_value,
+        ):
+            if isinstance(data, cls):
+                return data._partitions[part_idx][0]._data
+            if isinstance(data, pandas.Series):
+                return data[data_offset : data_offset + part_len]
+            # As mentioned above, this is required for Dask
+            if not single_wrap and callable(data):
+                return wrapper_put(data)
+            return (
+                data[data_offset : data_offset + part_len]
+                if is_list_like(data)
+                else data
+            )
+
+        parts = [p[0] for p in self._partitions]
+        lengths = self._get_lengths(parts, Axis.ROW_WISE)
```

Comment:
Why not like this? There could be already computed cache for them
```suggestion
        lengths = self.row_lengths
```